packages/transpiler/src/transformations/transform-constructor.ts

```diff
@@ -0,0 +1,94 @@
+import { getVarDeclarations, getNodeSources, getConstructor } from '../solc/ast-utils';
+import { Transformation } from '../transformation';
+import { buildSuperCallsForChain } from './build-super-calls-for-chain';
+import { ContractDefinition } from '../solc/ast-node';
+import { Artifact } from '../solc/artifact';
+
+function getVarInits(contractNode: ContractDefinition, source: string): string {
+  const varDeclarations = getVarDeclarations(contractNode);
+  return varDeclarations
+    .filter(vr => vr.value && !vr.constant)
+    .map(vr => {
+      const [, , varSource] = getNodeSources(vr, source);
+
+      const match = /(.*)(=.*)/.exec(varSource);
+      if (!match) throw new Error(`Can't find = in ${varSource}`);
+      return `\n        ${vr.name} ${match[2]};`;
+    })
+    .join('');
+}
+
+export function transformConstructor(
+  contractNode: ContractDefinition,
+  source: string,
+  contracts: string[],
+  contractsToArtifactsMap: Record<string, Artifact>,
+): Transformation[] {
+  const superCalls = buildSuperCallsForChain(contractNode, source, contracts, contractsToArtifactsMap);
+
+  const declarationInserts = getVarInits(contractNode, source);
+
+  const constructorNode = getConstructor(contractNode);
+
+  const isFullyImplemented = contractNode.fullyImplemented;
+
+  let removeConstructor = null;
+  let constructorBodySource = null;
+  let constructorParameterList = null;
+  let constructorArgsList = null;
+  if (constructorNode) {
+    constructorBodySource = getNodeSources(constructorNode.body, source)[2]
+      .slice(1)
+      .slice(0, -1);
+
+    constructorParameterList = getNodeSources(constructorNode.parameters, source)[2]
+      .slice(1)
+      .slice(0, -1);
+
+    const [start, len] = getNodeSources(constructorNode, source);
+
+    removeConstructor = {
+      start: start,
+      end: start + len,
+      text: '',
+    };
+
+    constructorArgsList = constructorNode.parameters.parameters.map(par => par.name).join(',');
+  }
+
+  constructorParameterList = constructorParameterList ? constructorParameterList : '';
+  const constructorParameterListWithComma = constructorParameterList ? `, ${constructorParameterList}` : '';
+  constructorBodySource = constructorBodySource ? constructorBodySource : '';
+  constructorArgsList = constructorArgsList ? `, ${constructorArgsList}` : '';
+
+  const initializeFuncText = isFullyImplemented
```

Comment:
> I'm assuming that isFullyImplemented means the contract is not abstract (in the solc 0.5 sense).
Correct.
Can't recall at that point but that is wrong because you can call `abstract` method on a child multiple times and that would be an error.

Reply:
Hmm. I've checked the code. It actually makes sense. Why would you want a public `initialize` method on an abstract class? The real constructor is inside `__init` method.

Reply:
> The real constructor is inside __init method.

Makes sense, since it's `__init` what you call from a derived contract